### PR TITLE
FEAT: add Azure ServiceBus Queue/Topic MessagePump registration shortcuts

### LINKED ISSUE #48: Provide simplified approach to wire up the Service Bus pump
Provide simplified version to use Service Bus pump which does not require you to use a func to get the connection string, but a secret key name instead:
```csharp
services.AddServiceBusTopicMessagePump<EmptyMessagePump>("Receive-All", "My-Secret-Name");
```

The documentation should clearly state that it will look it up by using `ISecretProvider` and not `IConfigurationProvider`.

### PR DESCRIPTION
**Pull request by stijnmoreels**
Closes #48

### CODE REVIEW

**Comment by tomkerkhove** [comment]
LGTM, one small suggestion would be to maybe add a quick test if easy to do.

**Comment by stijnmoreels** [comment]
> LGTM, one small suggestion would be to maybe add a quick test if easy to do.

Well, that was indeed something I was thinking about. Most of the extension methods in that file are never used. (except for the ones used in the dummy test projects). So, unless I add another project to verify this extension method?

**Comment by tomkerkhove** [comment]
We should provide some unit tests for them imo, just to see if they are wired up all fine or not - But ok to do via seperate issue if you want.

**Comment by stijnmoreels** [comment]
> We should provide some unit tests for them imo, just to see if they are wired up all fine or not - But ok to do via seperate issue if you want.

Ah, yes. That could be done.

File: src/Arcus.Messaging.Tests.Unit/ServiceBus/IServiceCollectionExtensionsTests.cs [hunk 1]
```diff
@@ -0,0 +1,53 @@
+﻿using Microsoft.Extensions.DependencyInjection;
+using Xunit;
+
+namespace Arcus.Messaging.Tests.Unit.ServiceBus
```
Comment (tomkerkhove) [inline src/Arcus.Messaging.Tests.Unit/ServiceBus/IServiceCollectionExtensionsTests.cs]: Shouldn't we also assert that it registers the correct lookup?

Comment (stijnmoreels) [reply src/Arcus.Messaging.Tests.Unit/ServiceBus/IServiceCollectionExtensionsTests.cs]: Hmm, how would you do that without making it an integration test?

Comment (tomkerkhove) [reply src/Arcus.Messaging.Tests.Unit/ServiceBus/IServiceCollectionExtensionsTests.cs]: Check the type that was registered in the collection and how it looks it up, if possible.

Or use mocking to see which one was called how many times?

I don't know, just thinking out loud.

Comment (stijnmoreels) [reply src/Arcus.Messaging.Tests.Unit/ServiceBus/IServiceCollectionExtensionsTests.cs]: Ah, now I see what you mean. I'll check that.

File: src/Arcus.Messaging.Tests.Unit/ServiceBus/IServiceCollectionExtensionsTests.cs [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+{
+    public class IServiceCollectionExtensionsTests
+    {
+        [Fact]
+        public async Task AddServiceBusTopicMessagePump_WithSubscriptionNameIndirectSecretProvider_WiresUpCorrectly()
+        {
+            // Arrange
+            var services = new ServiceCollection();
+            var spySecretProvider = new Mock<ISecretProvider>();
```
Comment (tomkerkhove) [inline src/Arcus.Messaging.Tests.Unit/ServiceBus/IServiceCollectionExtensionsTests.cs]: Spy 😂

**Comment by tomkerkhove** [comment]
Would be great to just update the docs as well, thanks!

File: src/Arcus.Messaging.Tests.Unit/ServiceBus/IServiceCollectionExtensionsTests.cs (hunk 2 above)
Comment (stijnmoreels) [reply src/Arcus.Messaging.Tests.Unit/ServiceBus/IServiceCollectionExtensionsTests.cs]: 😄 yeah, believe it or not; but that's the official terminology of xUnit (not the library but the pattern) testing practices 😃 
http://xunitpatterns.com/Test%20Spy.html

**Comment by stijnmoreels** [comment]
> Would be great to just update the docs as well, thanks!

Do I have this right? There's not docs for anything message pump related? That should maybe even has it's own issue/PR bc it's propably much.

**Comment by tomkerkhove** [comment]
Oh yeah my bad, that's actually on my list 😂

**Pull request merged**